controllers/azurejson_machinetemplate_controller.go

```diff
@@ -56,12 +59,34 @@ func (r *AzureJSONTemplateReconciler) SetupWithManager(ctx context.Context, mgr
 	)
 	defer done()
 
-	return ctrl.NewControllerManagedBy(mgr).
+	azureMachineTemplateMapper, err := util.ClusterToObjectsMapper(r.Client, &infrav1.AzureMachineTemplateList{}, mgr.GetScheme())
```

Comment:
This whole section of code down to `return nil` looks like it is repeated three times, one for each controller. Since they're in the same package, would it be worth refactoring this into a function they could share?

Reply:
I'm leaning towards leaving it as is. My understanding is that although the SetupWithManager() functions are similar in this circumstance, it is generally tailored to each type/controller (see the [AzureMachine controller](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azuremachine_controller.go#L74)) and and in the future we might want to changing it to be more specific.

Reply:
Agreed, and the potential function would take...looks like 5 or 6 parameters, so that feels a bit awkward.